providers/redfish/firmware.go

```diff
@@ -197,6 +197,71 @@ func (c *Conn) firmwareUpdateCompatible(ctx context.Context) (err error) {
 	return nil
 }
 
+// pipeReaderFakeSeeker wraps the io.PipeReader and implements the io.Seeker interface
+// to meet the API requirements for the Gofish client https://github.com/stmcginnis/gofish/blob/46b1b33645ed1802727dc4df28f5d3c3da722b15/client.go#L434
+//
+// The Gofish method linked does not currently perform seeks and so a PR will be suggested
+// to change the method signature to accept an io.Reader instead.
+type pipeReaderFakeSeeker struct {
+	*io.PipeReader
+}
+
+// Seek impelements the io.Seeker interface only to panic if called
+func (p pipeReaderFakeSeeker) Seek(offset int64, whence int) (int64, error) {
+	panic("Seek() not implemented for fake pipe reader seeker.")
+}
+
+// multipartPayloadSize prepares a temporary multipart form to determine the form size
+func multipartPayloadSize(payload []map[string]io.Reader) (int64, *bytes.Buffer, error) {
```

Comment:
this func seems rather wasteful, you go through the effort of creating a form only to throw it away to then create another one later in the caller of this func with almost duplicate code... why not just return the form + size from this func?

Reply:
The `multipartPayloadSize()` function copies the contents of the multipart form into a `bytes.Buffer{}` to determine the actual form size. 
The`runRequestWithMultipartPayload()` uses a `io.PipeWriter` which means the `io.Copy()` blocks until whenever the http request is actually being sent - to pipe the update file contents. 

So I don't see a trivial way to determine the Content-Length without building the multipart form once before actually sending it, I'm open to any ideas thought.

Reply:
Aw yeah I see that now. Nope I can't really think of another option w/o checking interfaces for Len and/or Size and maybe avoid the read. :(